File: app/src/main/java/io/inabsentia/superhangman/data/dao/IHighScoreDAO.java
package io.inabsentia.superhangman.data.dao;

import android.content.Context;

import java.util.List;

import io.inabsentia.superhangman.data.dto.HighScoreDTO;

public interface IHighScoreDAO {
    void add(HighScoreDTO dto) throws DALException;

    HighScoreDTO get(int id) throws DALException;

    List<HighScoreDTO> getAll() throws DALException;

    void update(HighScoreDTO dto) throws DALException;

    void remove(int id) throws DALException;

    void removeAll() throws DALException;

    void load(Context context) throws DALException;

    void save(Context context) throws DALException;
}

File: app/src/main/java/io/inabsentia/superhangman/data/dao/IHighScoreDAO.kt
package io.inabsentia.superhangman.data.dao

import android.content.Context

import io.inabsentia.superhangman.data.dto.HighScoreDTO

interface IHighScoreDAO {
    val all: List<HighScoreDTO>?

    @Throws(DALException::class)
    fun add(dto: HighScoreDTO)

    @Throws(DALException::class)
    operator fun get(id: Int): HighScoreDTO?

    @Throws(DALException::class)
    fun getCurrentHighScore(name: String): Int

    @Throws(DALException::class)
    fun update(dto: HighScoreDTO)

    @Throws(DALException::class)
    fun remove(id: Int)

    @Throws(DALException::class)
    fun removeAll(context: Context)

    @Throws(DALException::class)
    fun load(context: Context)

    @Throws(DALException::class)
    fun save(context: Context)
}